### PR TITLE
feat(productivity-suite): Add query param for piercing enabled

### PR DESCRIPTION
**Pull request by jculvey**

### CODE REVIEW

**Review by IgorMinar — APPROVED** [review]
LGTM! Thank you

**Review by petebacondarwin — CHANGES_REQUESTED** [review]

File: productivity-suite/app/legacy-app/index.html [hunk 1]
```diff
@@ -311,8 +311,9 @@
       const enablePiercingInput = document.getElementById('enable-piercing-input');
       enablePiercingInput.addEventListener('input', (event) => setPiercingEnablementCookie(event.target.checked));
 
-      const match = document.cookie.match(/piercingEnabled=(true|false)/);
-      const initialPiercingEnabledValue = !match ? null : match[1] === 'true' ? true : false;
+      const searchParams = new URLSearchParams(document.location.search)
+      const cookieMatch = document.cookie.match(/piercingEnabled=(true|false)/);
+      const initialPiercingEnabledValue = !cookieMatch ? searchParams.get("piercingEnabled") === 'true' : cookieMatch[1] === 'true' ? true : false;
```
[the remark this answers is not in the document]
Comment (dario-piotrowicz) [reply productivity-suite/app/legacy-app/index.html:316]: yeah I think they are available:
https://caniuse.com/?search=nullish
https://caniuse.com/mdn-javascript_operators_optional_chaining

[the remark this answers is not in the document]
Comment (dario-piotrowicz) [reply productivity-suite/app/legacy-app/index.html:316]: ah yeah you version's much better 🙂

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
As discussed in the meeting. The Gateway worker should be responsible for updating the cookie based on the query param, and then the client-side app can just read the cookie.

**Comment by dario-piotrowicz** [comment]
> ~As discussed in the meeting. The Gateway worker should be responsible for updating the cookie based on the query param, and then the client-side app can just read the cookie.~
> 
> Oh actually, I think that we don't want the cookie to change just because the search params overrides it... So this is probably fine, once we have fixed up the login in the client-side, right?

I had this conversation with James earlier, I think we do need the cookie to work because we want to have the checkbox checked on the browser, so we either override the cookie or find a different way to check the checkbox

Regarding the login, I thought your PR for it was all ready? is something still missing?

**Pull request closed without merging**